### YamlMime:ManagedReference
items:
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs
  id: DataGridViewDataErrorEventArgs
  children:
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  - System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  langs:
  - csharp
  name: DataGridViewDataErrorEventArgs
  nameWithType: DataGridViewDataErrorEventArgs
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs
  type: Class
  summary: "提供有关数据<xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;></xref>事件。"
  remarks: "处理<xref:System.Windows.Forms.DataGridView.DataError>事件可让你控制 （例如，通过外部数据源） 之外的代码引发的句柄异常。</xref:System.Windows.Forms.DataGridView.DataError> 使用<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A>属性来确定的状态<xref:System.Windows.Forms.DataGridView>时的异常。</xref:System.Windows.Forms.DataGridView> </xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Context%2A> 使用<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A>属性，以检索异常数据。</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> 如果你想要按其他事件处理程序处理异常，设置<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>属性`true`。</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>       <xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>和<xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A>属性通常指示发生数据错误该单元格。</xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex%2A> </xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A> 外部数据源错误时，但是，数据源可能无法提供发生错误的列。 在这种情况下，<xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>属性通常在出错时指示当前单元格的列。</xref:System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex%2A>"
  example:
  - "The following code example demonstrates how to respond to information provided by the DataGridViewDataErrorEventArgs class. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/t-system.windows.forms.d_56_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/t-system.windows.forms.d_56_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/t-system.windows.forms.d_56_1.vb)]"
  syntax:
    content: 'public class DataGridViewDataErrorEventArgs : System.Windows.Forms.DataGridViewCellCancelEventArgs'
  inheritance:
  - System.ComponentModel.CancelEventArgs
  - System.Windows.Forms.DataGridViewCellCancelEventArgs
  implements: []
  inheritedMembers:
  - System.Windows.Forms.DataGridViewCellCancelEventArgs.ColumnIndex
  - System.Windows.Forms.DataGridViewCellCancelEventArgs.RowIndex
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  id: '#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)'
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  type: Constructor
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "初始化的新实例<xref href=&quot;System.Windows.Forms.DataGridViewDataErrorEventArgs&quot;></xref>类。"
  syntax:
    content: public DataGridViewDataErrorEventArgs (Exception exception, int columnIndex, int rowIndex, System.Windows.Forms.DataGridViewDataErrorContexts context);
    parameters:
    - id: exception
      type: System.Exception
      description: "发生的异常。"
    - id: columnIndex
      type: System.Int32
      description: "引发的单元格的列索引<xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;> </xref>。"
    - id: rowIndex
      type: System.Int32
      description: "引发的单元格的行索引<xref href=&quot;System.Windows.Forms.DataGridView.DataError&quot;> </xref>。"
    - id: context
      type: System.Windows.Forms.DataGridViewDataErrorContexts
      description: "按位组合<xref href=&quot;System.Windows.Forms.DataGridViewDataErrorContexts&quot;></xref>值，该值指示发生了错误的上下文。"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  id: Context
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "获取有关状态的详细信息<xref href=&quot;System.Windows.Forms.DataGridView&quot;></xref>发生错误。"
  remarks: "成员<xref:System.Windows.Forms.DataGridViewDataErrorContexts>可能使用的按位组合枚举`OR`运算符来表示数据绑定的状态<xref:System.Windows.Forms.DataGridView>何时出现数据错误。</xref:System.Windows.Forms.DataGridView> </xref:System.Windows.Forms.DataGridViewDataErrorContexts> 例如，如果用户输入无效的单元格值 （例如，在需要日期的单元格中输入一个名称），然后选择不同的单元格，<xref:System.Windows.Forms.DataGridView>将尝试提交无效的单元格值。</xref:System.Windows.Forms.DataGridView> 提交失败时，<xref:System.Windows.Forms.DataGridView>将引发<xref:System.Windows.Forms.DataGridView.DataError>其上下文属性将具有的<xref:System.Windows.Forms.DataGridViewDataErrorContexts>和<xref:System.Windows.Forms.DataGridViewDataErrorContexts>。</xref:System.Windows.Forms.DataGridViewDataErrorContexts></xref:System.Windows.Forms.DataGridViewDataErrorContexts>值的事件</xref:System.Windows.Forms.DataGridView.DataError></xref:System.Windows.Forms.DataGridView>"
  example:
  - "The following code example demonstrates how to investigate the error context. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_134_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_134_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_134_1.vb)]"
  syntax:
    content: public System.Windows.Forms.DataGridViewDataErrorContexts Context { get; }
    return:
      type: System.Windows.Forms.DataGridViewDataErrorContexts
      description: "按位组合<xref href=&quot;System.Windows.Forms.DataGridViewDataErrorContexts&quot;></xref>指定发生错误的上下文的值。"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  id: Exception
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "获取表示错误的异常。"
  remarks: "若要处理的异常的多个<xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>，通过设置引发异常，在处理程序末尾时再次<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A>属性`true`。</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException%2A> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>"
  example:
  - "The following code example demonstrates checking the Exception property. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_249_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_249_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_249_1.vb)]"
  syntax:
    content: public Exception Exception { get; }
    return:
      type: System.Exception
      description: "<xref:System.Exception>表示的错误。</xref:System.Exception>"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*
  exceptions: []
  platform:
  - net462
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  id: ThrowException
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  langs:
  - csharp
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  type: Property
  assemblies:
  - System.Windows.Forms
  namespace: System.Windows.Forms
  summary: "获取或设置一个值，该值指示是否后引发该异常<xref href=&quot;System.Windows.Forms.DataGridViewDataErrorEventHandler&quot;></xref>委托是完成它。"
  remarks: "将此属性设置为`false`如果<xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>所表示的异常已处理<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A>属性，而你不想将传播到另一个异常<xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>，可能会处理<xref:System.Windows.Forms.DataGridView.DataError>事件。</xref:System.Windows.Forms.DataGridView.DataError> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler> </xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception%2A> </xref:System.Windows.Forms.DataGridViewDataErrorEventHandler>       出于调试目的，它可用于传播异常，并检查错误的环境相关的信息的堆栈跟踪。"
  example:
  - "The following code example demonstrates using the ThrowException property to indicate that the exception should not be thrown. This example is part of a larger example available in the <xref:System.Windows.Forms.DataGridViewComboBoxColumn> class overview topic.  \n  \n [!code-cpp[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/cpp/p-system.windows.forms.d_202_1.cpp)]\n [!code-cs[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/csharp/p-system.windows.forms.d_202_1.cs)]\n [!code-vb[System.Windows.Forms.DataGridViewColumn_AllColumns_Bound_Employees#50](~/add/codesnippet/visualbasic/p-system.windows.forms.d_202_1.vb)]"
  syntax:
    content: public bool ThrowException { get; set; }
    return:
      type: System.Boolean
      description: "<xref uid=&quot;langword_csharp_true&quot; name=&quot;true&quot; href=&quot;&quot;></xref>如果应引发异常;否则为<xref uid=&quot;langword_csharp_false&quot; name=&quot;false&quot; href=&quot;&quot;> </xref>。 默认值是<xref uid=&quot;langword_csharp_false&quot; name=&quot;false&quot; href=&quot;&quot;> </xref>。"
  overload: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException*
  exceptions:
  - type: System.ArgumentException
    commentId: T:System.ArgumentException
    description: "当此属性设置为<xref uid=&quot;langword_csharp_true&quot; name=&quot;true&quot; href=&quot;&quot;> </xref>、<xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*>属性值是<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>。</xref:System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*>"
  platform:
  - net462
references:
- uid: System.Windows.Forms.DataGridViewCellCancelEventArgs
  isExternal: false
  name: System.Windows.Forms.DataGridViewCellCancelEventArgs
- uid: System.ArgumentException
  isExternal: true
  name: System.ArgumentException
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor(System.Exception,System.Int32,System.Int32,System.Windows.Forms.DataGridViewDataErrorContexts)
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs(Exception,Int32,Int32,DataGridViewDataErrorContexts)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Int32
  parent: System
  isExternal: true
  name: Int32
  nameWithType: Int32
  fullName: System.Int32
- uid: System.Windows.Forms.DataGridViewDataErrorContexts
  parent: System.Windows.Forms
  isExternal: false
  name: DataGridViewDataErrorContexts
  nameWithType: DataGridViewDataErrorContexts
  fullName: System.Windows.Forms.DataGridViewDataErrorContexts
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
  fullName: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException
- uid: System.Boolean
  parent: System
  isExternal: true
  name: Boolean
  nameWithType: Boolean
  fullName: System.Boolean
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.#ctor*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: DataGridViewDataErrorEventArgs
  nameWithType: DataGridViewDataErrorEventArgs.DataGridViewDataErrorEventArgs
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Context*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Context
  nameWithType: DataGridViewDataErrorEventArgs.Context
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.Exception*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: Exception
  nameWithType: DataGridViewDataErrorEventArgs.Exception
- uid: System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException*
  parent: System.Windows.Forms.DataGridViewDataErrorEventArgs
  isExternal: false
  name: ThrowException
  nameWithType: DataGridViewDataErrorEventArgs.ThrowException
